articles/cosmos-db/create-mongodb-golang.md

```diff
@@ -0,0 +1,279 @@
+
+4. Click **Create** to create the account.
+
+5. On the toolbar, click **Notifications** to monitor the deployment process.
+
+![](https://cdn-images-1.medium.com/max/800/1*WTGVkPwkSTxjUzKm2u09Eg.png)
+<span class="figcaption_hack">Check Progress of Azure Cosmos DB Deployment</span>
+
+6. Click on **golang-couch** resources.
+
+7. Get connection string information which will be required by client
+applications.
+
+![](https://cdn-images-1.medium.com/max/800/1*XoslPMsWsv5vqK2vhYAJPw.png)
+<span class="figcaption_hack">Connection information of MongoDB</span>
+
+### Setting up your application
+
+It’s time to make our hands dirty. Open your favorite editor (Gogland, VS Code
+or Atom). For this article, I will use Gogland editor.
+
+1.  Create folder CosmosDBAcces folder inside GOROOT\src folder
+1.  Run below command to get mgo package
+
+    go get gopkg.in/mgo.v2
+
+### Connect to an Azure Cosmos DB account
+
+[mgo](http://labix.org/mgo) (pronounced as *mango*) is a
+[MongoDB](http://www.mongodb.org/) driver for the [Go
+language](http://golang.org/) that implements a rich and well tested selection
+of features under a very simple API following standard Go idioms.
+
+Azure Cosmos DB uses latest version of MongoDB v3.2.0. with SSL enabled.
+
+To check MongoDb version go to Azure portal and get MongoDB Shell Tab and copy
+mongo command
+
+![](https://cdn-images-1.medium.com/max/800/1*voTl_TbS7Wuld2Q4CBLFXQ.png)
+<span class="figcaption_hack">Azure Cosmos MongoDB Shell</span>
+
+Open command prompt and run command. Once the connection is established, run
+below command to check database version.
+
+    db.version()
+
+![](https://cdn-images-1.medium.com/max/800/1*OJwDs_lG45z_K3rzw2_RuQ.png)
+
+Officially, mgo does not support MongoDB 3.2 version. There is solution to
+ignore server certificate validation by mgo. Please refer blog [Connect to
+MongoDB 3.2 on Compose from
+Golang](https://www.compose.com/articles/connect-to-mongo-3-2-on-compose-from-golang/)
+by [Hays Hutton](about:invalid#zSoyz)
+
+To get started we have to configure transport layer security (tls). We have to
+make sure that client Certificate Authority ignore SSL validation. While this
+does open up some attack vectors, it is is better than no SSL. Here are the
+comments directly from the source code of the **tls.Config** struct for the
+particular boolean we need to set:
+
+
+The following code snippet to connect with MongoDB with GO app. DialInfo holds
+options for establishing a session with a MongoDB cluster.
+
+```go
+tlsConfig := &tls.Config{}
+
+// InsecureSkipVerify controls whether a client verifies the
+// server's certificate chain and host name.
+// If InsecureSkipVerify is true, TLS accepts any certificate
+// presented by the server and any host name in that certificate.
+// In this mode, TLS is susceptible to man-in-the-middle attacks.
+// This should be used only for testing.
+tlsConfig.InsecureSkipVerify = true
```

Comment:
Based on my understanding, the tls package will validate and trust all the ROOTCA's which are trusted by the operation system.  As Cosmos DB certificate is issued by BaltimoreRootCA, this line should not be required while connection to a real Cosmos DB account  (it is required while developing against the local emulator and the comment should be updated).